File: inclass-004.py
studentScore = []
studentName = []
studentMAX, studentMIN = 0, 0
MAX, MIN, SUM, average = 0, 999, 0, 0
length = int(input("How many student in class: "))

while(length < 1):
    #check length if less than 1
    length = int(input("Invalid number. Please input more than 1: "))

for i in range(length):
    #input name and score
    name = str(input("Please enter student name: "))
    studentName.append(name)
    score = float(input("Please enter %s score: " %name))
    studentScore.append(score)
    
    #check score
    if studentScore[i] > MAX:
        MAX = studentScore[i]
        studentMAX = studentName[i]
    if studentScore[i] < MIN:
        MIN = studentScore[i]
        studentMIN = studentName[i]
    SUM = SUM + score
    
#find average
average = SUM / length

print("Highest score is %s have %.2f score" %(studentMAX, MAX))
print("Lowest score is %s have %.2f score" %(studentMIN, MIN))
print("Average score: %.2f" %(average))

File: inclass-004.cpp
#include <stdio.h>
#include <string.h>

int main(){
    int length = 0;
    float score[length], average, sum = 0, min = 999, max = 0;
    char studentName[50], studentMax[50], studentMin[50];
    
    printf("Please input number of student in class: ");
    scanf("%d", &length);
    
    while(length < 1){
		printf("Invalid number. Please input again: ");
		scanf("%d", &length);
	}
    
    for(int i = 0; i < length; i++){
        printf("Input student name %d: ", i+1);
        scanf("%s", studentName);
        printf("Input %s score %d: ", studentName, i+1);
        scanf("%f", &score[i]);
        
        if(score[i] > max){
            max = score[i];
            strcpy(studentMax, studentName);
        }
        if(score[i] < max){
            min = score[i];
            strcpy(studentMin, studentName);
        }
        sum += score[i];
    }
    
    average = sum / length;
    
    printf("Maximun score is: %s  ", studentMax);
    printf("have %.2f score\n", max);
    printf("Minimun score is: %s  ", studentMin);
    printf("have %.2f score\n", min);
    printf("Average score %.2f\n", average);
    
}
